ts/components/session/settings/SessionSettingsHeader.tsx

```diff
@@ -6,19 +6,18 @@ import { SessionButton } from '../SessionButton';
 
 interface Props extends SettingsViewProps {
   showLinkDeviceButton: boolean | null;
-  disableLinkDeviceButton: boolean | null;
+  isSecondaryDevice: boolean;
```

Comment:
having both of these has a weird smell to me. But I'm not clear on if there's 4 states here or just 2 (false/true && true/false)

Reply:
no thee is 4 states possible. 
* the showlink is used to completely hide the button when the settings screen is still locked by a password
* isSecondaryDevice is used to disable the button only when we are a secondary device